packages/platform-browser/src/IndexedDbQueryBuilder.ts

```diff
@@ -0,0 +1,570 @@
+    IndexedDbTable.IndexedDbTable.AutoIncrement<
+      IndexedDbTable.IndexedDbTable.WithName<
+        IndexedDbVersion.IndexedDbVersion.Tables<Source>,
+        Table
+      >
+    > extends false ? /** keyPath: string, autoIncrement: false */ SourceTableSchemaType<Source, Table> :
+    & /** keyPath: string, autoIncrement: true */ Omit<
+      SourceTableSchemaType<Source, Table>,
+      IndexedDbTable.IndexedDbTable.KeyPath<
+        IndexedDbTable.IndexedDbTable.WithName<
+          IndexedDbVersion.IndexedDbVersion.Tables<Source>,
+          Table
+        >
+      >
+    >
+    & {
+      [
+        K in IndexedDbTable.IndexedDbTable.KeyPath<
+          IndexedDbTable.IndexedDbTable.WithName<
+            IndexedDbVersion.IndexedDbVersion.Tables<Source>,
+            Table
+          >
+        >
+      ]?: Pick<
+        Schema.Schema.Type<
+          IndexedDbVersion.IndexedDbVersion.SchemaWithName<Source, Table>
+        >,
+        IndexedDbTable.IndexedDbTable.KeyPath<
+          IndexedDbTable.IndexedDbTable.WithName<
+            IndexedDbVersion.IndexedDbVersion.Tables<Source>,
+            Table
+          >
+        >
+      >[K]
+    }
+
+  /**
+   * @since 1.0.0
+   * @category models
+   */
+  export interface From<
+    Source extends IndexedDbVersion.IndexedDbVersion.AnyWithProps = never,
+    Table extends IndexedDbTable.IndexedDbTable.TableName<
+      IndexedDbVersion.IndexedDbVersion.Tables<Source>
+    > = never
+  > {
+    readonly [TypeId]: TypeId
+    readonly tables: HashMap.HashMap<string, IndexedDbVersion.IndexedDbVersion.Tables<Source>>
+    readonly table: Table
+    readonly database: globalThis.IDBDatabase
+    readonly IDBKeyRange: typeof globalThis.IDBKeyRange
+    readonly transaction?: globalThis.IDBTransaction
+
+    readonly select: {
+      <Index extends IndexFromTable<Source, Table>>(index: Index): Select<Source, Table, Index>
+      (): Select<Source, Table, never>
+    }
+
+    readonly count: {
+      <Index extends IndexFromTable<Source, Table>>(index: Index): Count<Source, Table, Index>
+      (): Count<Source, Table, never>
+    }
+
+    readonly delete: {
+      <Index extends IndexFromTable<Source, Table>>(index: Index): Delete<Source, Table, Index>
+      (): Delete<Source, Table, never>
+    }
+
+    readonly insert: (value: ModifyWithKey<Source, Table>) => Modify<Source, Table>
+    readonly insertAll: (values: Array<ModifyWithKey<Source, Table>>) => ModifyAll<Source, Table>
+    readonly upsert: (value: ModifyWithKey<Source, Table>) => Modify<Source, Table>
+    readonly upsertAll: (values: Array<ModifyWithKey<Source, Table>>) => ModifyAll<Source, Table>
+    readonly clear: Clear<Source, Table>
+  }
+
+  /**
+   * @since 1.0.0
+   * @category models
+   */
+  export interface ClearAll<
+    Source extends IndexedDbVersion.IndexedDbVersion.AnyWithProps = never
+  > {
+    [Symbol.iterator](): Effect.EffectGenerator<Effect.Effect<void>>
```

Comment:
What is the Symbol.iterator for?

Reply:
I simplified `clearAll` and `clear` to just return an `Effect` instead of an intermediate `interface`